UnchainedLauncher.Core/Extensions/EitherExtensions.cs

```
@@ -62,5 +62,24 @@ public static EitherAsync<L, R> BindTapLeft<L, R, L2>(this EitherAsync<L, R> eit
                 return f(l).MapLeft(_ => l);
             });
         }
+
+        public static EitherAsync<L, R> ToEitherAsync<L, R>(this Either<L, R> either) =>
+            either.Match(EitherAsync<L, R>.Right, EitherAsync<L, R>.Left);
+
+        public static Either<TL, Unit> AggregateBind<TL>(this IEnumerable<Either<TL, Unit>> eithers) {
```

A better name for this would be `FirstLeft`, and on that note I think a better return value would be `Option<TL>`, and also a potentially even better name might be `Lefts` and return `IEnumerable<TL>`.  

When you have an `Either<L, R>`, the work has already been done. So if you only return the first error, you're potentially discarding other errors

---

Maybe the solution is to explicitly use List or some eagerly evaluated collection

---

I was indeed thinking of this as a lazy-evaluated bind where if one failed, the rest aren't attempted. My perception of the monad ops has been "here's a function I want you to add to the chain representing this monad state. Execute it if you ever need the result" but maybe that's not quite what's actually happening. From a type perspective it doesn't matter if you execute it immediately vs only when you reach a junction like a Match or Tap, but it definitely matters in terms of what *actually* happens (and probably how hellish it is to debug). Surely if we wrapped our IO into the proper effect types instead of having it as side effects then it would mean the results coincide with the types, but that's kind of tedious to do because the returns and mutations all depend on the actual result of the IO going through vs failing.

As far as error reporting goes, it's probably fine; all we *really* need to know is "something failed, and here's the neighborhood" because it's likely that fixing one thing impeding a file op will fix all the failed file ops. I will change it to a Lefts()-type call though, because that's ultimately what I'm actually doing.